nisystemlink/cross-service/GenericApi.py

```diff
@@ -0,0 +1,1040 @@
+                                    }
+
+                for found_channel in found_dict.keys():
+                    channels.remove(found_channel)
+                return_dict.update(found_dict)
+        if len(channels) > 0:
+            error_message = "Missing the following channels: "
+            for channel in channels:
+                error_message.join(f"{channel}, ")
+            error_message = error_message[:-2]
+            raise KeyError(error_message)
+        return return_dict
+
+    def get_channel_meta(
+        self, channel: List[str] = None, key: List[str] = None
+    ) -> Dict[str, Dict[str, str]]:
+        """Return some or all of the channel metadata.
+
+        Channel data may include things like units or other identifying characterstics
+
+        :param channel: a list describing the channels (columns) to return data for
+        :param key: a list describing the keys to return values for
+        """
+        raise NotImplementedError()
+        if key:
+
+            # return specified key-value pairs as a dictionary, with in a dictionary
+            # where the key represents the channel name
+            return ""
+        else:
+            # return all key-value pairs as a dictionary, with in a dictionary
+            # where the key represents the channel name
+            return ""
+
+    async def get_synonym_table(self, file_name: str = "synonyms.csv") -> Dict[str, str]:
+        """Get the synonym lookup table as a dictionary."""
+        property_query = fis_models.PropertyQuery("Name", "EQUAL", file_name)
+        fis_query = fis_models.QueryAvailableFilesRequest(properties_query=[property_query])
+        file_list = await self._fis_handle.query_available_files(query=fis_query)
+        properties = file_list.available_files[0].properties
+        return properties  # todo: Need a test for this
+
+    async def close_connections(self):
+        """Close all open session handles."""
+        await self._tm_result_handle.api_client.close()
+        await self._fis_handle.api_client.close()
+
+    async def _post_data_to_dfs(self, body, frame_id, max_tries=5):
+        num_trys = 0
+        upload_success = False
+        local_http_conn = http.client.HTTPSConnection(self._sl_uri + ":443")
+        while not upload_success and num_trys < max_tries:
+            local_http_conn.request(
+                "POST",
+                "/nidataframe/v1/tables/{}/data".format(frame_id),
+                body=json.dumps(body),
+                headers=self._headers,
+            )
+            r = local_http_conn.getresponse()
+            if r.status == 204:
+                upload_success = True
+            else:
+                num_trys += 1
+                print(r.status)
+                print("Upload attempt {} failed".format(num_trys))
+        local_http_conn.close()
+        return upload_success
+
+    async def get_test_id_from_file_id(self, file_id: str):
```

Comment:
This gets a hard-coded property from a file - don't think it should be in this package